### PR TITLE
Add configs to system menu option

### PR DESCRIPTION
**Pull request by oscfdezdz**
These configs work properly with the solution proposed in https://github.com/daitj/gnome-display-brightness-ddcutil/issues/41 but I did not apply it here because it seems like a workaround.

853e3b6f0ce37a473bf55ff8d245e784f0217535:

imagen

imagen

06c683bdd31d088ad5b470bcf848d7cb2f353c47:

I could only get the system menu layout to update by logging out.

Current behavior for me:
imagen

imagen

The first brightness slider is from the integrated display.

### CODE REVIEW

**Comment by oscfdezdz** [comment]
Do not merge this one yet, please

**Comment by oscfdezdz** [comment]
It should be good now!

**Comment by daitj** [comment]
Have you tested the scroll and keyboard event which are attached to the indicator icon. If they dont work, you should be able to hide the indicator by setting its visibility option, instead of not using them at all.

About the issue #41, according to [night-light-slider](https://codeberg.org/kiyui/gnome-shell-night-light-slider-extension/src/branch/main/src/extension.js) extension's implementation, the menu items should be destoryed.

**Comment by oscfdezdz** [comment]
> Have you tested the scroll and keyboard event which are attached to the indicator icon. If they dont work, you should be able to hide the indicator by setting its visibility option, instead of not using them at all.

Yes, that was the last minute change, but I will check the visibility option which should be best here.

> About the issue #41, according to [night-light-slider](https://codeberg.org/kiyui/gnome-shell-night-light-slider-extension/src/branch/main/src/extension.js) extension's implementation, the menu items should be destoryed.

Doesn't that already happen when reloading the extension? https://github.com/daitj/gnome-display-brightness-ddcutil/blob/bb2f0ffd4ee0726a6fa4f4f6b7324960b2c095a9/display-brightness-ddcutil%40themightydeity.github.com/extension.js#L130

**Comment by daitj** [comment]
As far as I have understood, `mainMenu.destory` just destorys the indicator, but all the menu items are actually attached to the system menu area, so they have to be destroyed separately. I will try to sort this out. 

About the indicators visiblity you can check [here](https://github.com/GNOME/gnome-shell/blob/main/js/ui/panelMenu.js), line 204-207 on how indicator is initialized.

File: display-brightness-ddcutil@themightydeity.github.com/indicator.js [hunk 1]
```diff
@@ -84,12 +84,12 @@ var SystemMenuBrightnessMenu = GObject.registerClass({
...
 }, class SystemMenuBrightnessMenu extends PanelMenu.SystemIndicator {
-    _init() {
+    _init(settings) {
         super._init();
+        this._valueSliders = [];
         this._indicator = this._addIndicator();
         this._indicator.icon_name = 'display-brightness-symbolic';
-        this._valueSliders = [];
-
+        this._indicator.visible = !settings.get_boolean('hide-system-indicator');
```
Comment (oscfdezdz) [inline display-brightness-ddcutil@themightydeity.github.com/indicator.js:92]: It is using `visible` property now.

**Comment by daitj** [comment]
Could you rebase your changes, I had to fix some issues which resulted in some conflicts. 

Thanks.

**Comment by oscfdezdz** [comment]
With the latest changes in master it is no longer necessary to log out to change the position in the system menu. Now the switch could be a spin button as I had originally intended, but I will leave it that way because there are no more cases where the position is different depending on the number of items in the menu.

**Comment by daitj** [comment]
I think it would be better to have number spin instead of toggle for position, I can leave the PR open for now. You can fix this when you can. 

If you do not have time, you can remove that position code for now from this PR and just leave the code for hiding the indicator.

**Comment by oscfdezdz** [comment]
imagen

The minimum and maximum menu positions are 0 and 13 according to the tests I have done.

**Pull request merged**